misc/hamming_distance2.c

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * @author [BenSt099](https://github.com/BenSt099)
+ * @brief Computes the [Hamming distance](https://en.wikipedia.org/wiki/Hamming_distance)
+ * of two Integer-Arrays
+ * @details
+ * In information theory, the Hamming distance between two strings of
+ * equal length is the number of positions at which the corresponding symbols
+ * are different.
+ * @see  
+ */
+
+#include <assert.h>  /// for assert
+#include <stdio.h>   /// for IO operations
+
+/**
+ * @brief Function that calculates the Hamming distance between two Integer-arrays
+ * @param param1 int* Pointer for Hamming distance
+ * @param param2 int  length of the two arrays
+ * @param param3 int* first int-array
+ * @param param4 int* second int-array
+ * @returns Pointer to Hamming distance
+ */
+void hamming_distance(int* h_dist, int length, int* s1, int* s2) {
```

Comment:
why not use int* as the return?
```suggestion
int* hamming_distance(int length, int* s1, int* s2) {
```